soc/xtensa/intel_adsp/cavs/gdbstub.c

```diff
@@ -0,0 +1,1394 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <xtensa-asm2-context.h>
+
+static struct xtensa_register gdb_reg_list[] = {
+/* XT-GDB has different register definitions than Zephyr toolchain GDB */
+#ifdef CONFIG_ADSP_XT_GDB
+	{
+		/* PC */
+		.idx = 32,
+		.regno = 0x0020,
+		.byte_size = 4,
+		.gpkt_offset = 0,
+		.stack_offset = BSA_PC_OFF,
+	},
```

Comment:
Gotta be honest: not really loving the format chosen here.  These are HUGE arrays, and AFAICT the only values that changes between the structs are the idx/regno and packet offset fields, right?  The other stuff is all architectural.  Maybe we could come up with a cleaner way to express this, maybe by generating the array somehow?  This seems like it's going to be extremely difficult to maintain as the tools evolve.  Note for example that the Xtensa HiFi SIMD registers are likely to need support, every new Xtensa core has some different mix of SRs (e.g. more MISC registers, more exception levels), etc...